python/semantic_kernel/connectors/memory/postgres/postgres_memory_store.py

```diff
@@ -0,0 +1,564 @@
+    ) -> None:
+        """Initializes a new instance of the PostgresMemoryStore class.
+
+        Arguments:
+            connection_string {str} -- The connection string to the Postgres database.\n
+            default_dimensionality {int} -- The default dimensionality of the embeddings.\n
+            min_pool {int} -- The minimum number of connections in the connection pool.\n
+            max_pool {int} -- The maximum number of connections in the connection pool.\n
+            schema {str} -- The schema to use. (default: {"public"})\n
+            timezone_offset {Optional[str]} -- The timezone offset to use. (default: {None})
+            Expected format '-7:00'. Uses the local timezone offset when not provided.\n
+            logger {Optional[Logger]} -- The logger to use. (default: {None})
+        """
+        if default_dimensionality > MAX_DIMENSIONALITY:
+            raise ValueError(
+                f"Dimensionality of {default_dimensionality} exceeds "
+                + f"the maximum allowed value of {MAX_DIMENSIONALITY}."
+            )
+        if default_dimensionality <= 0:
+            raise ValueError("Dimensionality must be a positive integer. ")
+
+        if timezone_offset is None:
+            t = time.time()
+
+            if time.localtime(t).tm_isdst and time.daylight:
+                float_offset = -time.altzone / 60 / 60
+            else:
+                float_offset = -time.timezone / 60 / 60
+
+            self._timezone_offset = (
+                str(float_offset).replace(".", ":")[:5]
+                if float_offset < 0
+                else str(float_offset).replace(".", ":")[:4]
+            )
+
+        self._connection_string = connection_string
+        self._default_dimensionality = default_dimensionality
+        self._connection_pool = ConnectionPool(
+            self._connection_string, min_size=min_pool, max_size=max_pool
+        )
+        self._schema = schema
+        atexit.register(self._connection_pool.close)
+        self._logger = logger or NullLogger()
+
+    async def create_collection_async(
+        self,
+        collection_name: str,
+        dimension_num: Optional[int] = None,
+    ) -> None:
+        """Creates a new collection.
+
+        Arguments:
+            collection_name {str} -- The name of the collection to create.\n
+            dimension_num {Optional[int]} -- The dimensionality of the embeddings. (default: {None})
+            Uses the default dimensionality when not provided
+
+        Returns:
+            None
+        """
+        if dimension_num is None:
+            dimension_num = self._default_dimensionality
+        if dimension_num > MAX_DIMENSIONALITY:
+            raise ValueError(
+                f"Dimensionality of {dimension_num} exceeds "
+                + f"the maximum allowed value of {MAX_DIMENSIONALITY}."
+            )
+        if dimension_num <= 0:
+            raise ValueError("Dimensionality must be a positive integer. ")
```

Comment:
Yeah this one is kind of tough for me since most of the memory store backends require a dimensionality at time of collection creation. I understand that the `MemoryStoreBase` doesn't have it as a parameter and as such the Kernel will not pass anything for it but I wonder if that should be the case. 

I wonder if there is a use case in which a consumer would interface with 2 embedding services (that return embeddings with different dimensionality). As the memory store stands now, they would need two instances of a memory store to support both dimensionalities, which I dont know if the Kernel even supports.

Food for thought.

Regardless, I can move the check into a single method but will maintain the parameter on the `create_collection_async` method for the time being